File: python/largest-series-product/largest_series_product.py
def largest_product(series, size):
    if size < 1:
        raise ValueError("span must not be negative")
    if len(series) < size:
        raise ValueError("span must be smaller than string length")
    if series.isdigit() == False:
        raise ValueError("digits input must only contain digits")

    max = 0
    
    for idx in range(0, len(series)-size+1, 1):
        num = 1
        for p in range(0, size):
            num *= int(series[idx+p])
        if num > max:
            max = num

    return max

File: javascript/largest-series-product/largest-series-product.js
export const largestProduct = (digits, span) => {
  if (span < 1)
    throw new Error("Span must be greater than zero")
  if (digits.length < span)
    throw new Error("Span must be smaller than string length")
  if (/^\d+$/.test(digits) === false)
    throw new Error("Digits input must only contain digits")

  let max = 0
  for (let i = 0; i <= digits.length - span; i++) {
    let num = digits[i]

    for (let j = 1; j < span; j++)
      num *= digits[i+j]

    max = num > max ? num : max
  }

  return max
}
